fairseq2n/src/fairseq2n/data/concat_data_source.cc

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include "fairseq2n/data/concat_data_source.h"
+#include <vector>
+
+namespace fairseq2n::detail {
+
+concat_data_source::concat_data_source(
```

Comment:
Nit: In C++ files, we typically break lines at 100 chars. I haven't counted, but looks like the `pipelines` parameter can be on the same line here.